base/special/cbrt.jl

```diff
@@ -0,0 +1,144 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+# Float32/Float64 based on C implementations from FDLIBM (http://www.netlib.org/fdlibm/)
+# and FreeBSD:
+#
+## ====================================================
+## Copyright (C) 1993 by Sun Microsystems, Inc. All rights reserved.
+##
+## Developed at SunPro, a Sun Microsystems, Inc. business.
+## Permission to use, copy, modify, and distribute this
+## software is freely granted, provided that this notice
+## is preserved.
+## ====================================================
+## Conversion to float by Ian Lance Taylor, Cygnus Support, ian@cygnus.com.
+## Debugged and optimized by Bruce D. Evans.
+
+"""
+    cbrt(x::Real)
+
+Return the cube root of `x`, i.e. ``x^{1/3}``. Negative values are accepted
+(returning the negative real root when ``x < 0``).
+
+The prefix operator `∛` is equivalent to `cbrt`.
+
+```jldoctest
+julia> cbrt(big(27))
+3.0
+```
+"""
+cbrt(x::Real) = cbrt(float(x))
+cbrt(x::AbstractFloat) = x < 0 ? -(-x)^(1//3) : x^(1//3)
+
+"""
+    _approx_cbrt(x)
+
+Approximate `cbrt` to 5 bits precision
+
+    cbrt(2^e * (1+m)) ≈ 2^(e÷3) * (1 + (e%3+m)÷3)
+
+where:
+ - `e` is integral and >= 0
+ - `m` is real and in [0, 1),
+ - `÷` is integer division
+ - `%` is integer remainder
+
+The RHS is always >= the LHS and has a maximum relative error of about 1 in 16.
+Adding a bias of -0.03306235651 to the `(e%3+m)÷3` term reduces the error to about 1 in
+32.
+
+With the IEEE floating point representation, for finite positive normal values, ordinary
+integer divison of the value in bits magically gives almost exactly the RHS of the above
+provided we first subtract the exponent bias and later add it back.  We do the
+subtraction virtually to keep e >= 0 so that ordinary integer division rounds towards
+minus infinity; this is also efficient. All operations can be done in 32-bit.
+
+These implementations assume that NaNs, infinities and zeros have already been filtered.
+"""
+@inline function _approx_cbrt(x::T) where {T<:Union{Float32,Float64}}
+    adj = exponent_bias(T)*2/3 - 0.03306235651
+    k = significand_bits(T) - (8*sizeof(T) - 32)
+
+    u = highword(x) & 0x7fff_ffff
+    if u >= Base.Math.highword(realmin(T))
+        v = div(u, UInt32(3)) + floor(UInt32, adj * 2.0^k)
+    else
+        # subnormal
+        x *= maxintfloat(T)
+        adj -= exponent(maxintfloat(T))/3
```

Comment:
should we multiply by `T(0.333333333333333333)` here instead of dividing by `3`?

Reply:
Again, is a constant, so should be evaluated at compile time.